www/js/app.js

```diff
@@ -998,16 +1001,47 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
             document.getElementById('articleContent').contentWindow.focus();
             $("#searchingArticles").show();
             if (dirEntry.isRedirect()) {
-                selectedArchive.resolveRedirect(dirEntry, readArticle);
+                // Update expectedArticleURLToBeDisplayed before redirect. 
+                selectedArchive.resolveRedirect(dirEntry, function (resolvedDirEntry) {
+                    updatedExpectedURLOnRedirect(resolvedDirEntry, readArticle);
```

Comment:
That should work (I did not have time to test for now, sorry).
On the other hand, I think it could be simplified by removing the updatedExpectedURLOnRedirect function (which was fine), and move the `expectedArticleURLToBeDisplayed = dirEntry.namespace + "/" + dirEntry.url;` at the beginning of `readArticle`.
It should replace several places where you call that function or do the same : lines 1006, 1010, 1103, 1540 and 1563.

Reply:
This is a good point. I use a helper function cause it is easier for me to log the debug messages while testing.